api/src/pcapi/core/subscription/api.py

```diff
@@ -495,48 +641,27 @@ def _get_activable_identity_check(
     return None
 
 
-def _has_completed_other_steps(user: users_models.User, activable_eligibility: users_models.EligibilityType) -> bool:
-    if _should_validate_phone(user, activable_eligibility):
-        return False
-
-    subscription_item = get_user_profiling_subscription_item(user, activable_eligibility)
-    if subscription_item.status in (models.SubscriptionItemStatus.TODO, models.SubscriptionItemStatus.KO):
-        return False
-
-    profile_completion = get_profile_completion_subscription_item(user, activable_eligibility)
-    if profile_completion.status != models.SubscriptionItemStatus.OK:
-        return False
-
-    if not fraud_api.has_performed_honor_statement(user, activable_eligibility):
-        return False
-
-    return True
-
-
 def activate_beneficiary_if_no_missing_step(user: users_models.User) -> bool:
-    if fraud_repository.has_admin_ko_review(user):
-        return False
+    subscription_state = get_user_subscription_state(user)
 
-    activable_fraud_check_and_eligibility = _get_activable_identity_check(user)
-
-    if not activable_fraud_check_and_eligibility:
+    if not subscription_state.is_activable or not subscription_state.fraud_check:
         return False
```

Comment:
la fonction vérifie plein de condition et retourne `False` dès la première vérification

je ne comprends pas pourquoi il y a un `or` pour cette condition

```suggestion
    if not subscription_state.is_activable:
        return False

    if not subscription_state.fraud_check:
        return False
```